xen/common/device_tree.c

```diff
@@ -1994,6 +1994,8 @@ static void __init __unflatten_device_tree(const void *fdt,
     *allnextp = NULL;
 
     dt_dprintk(" <- unflatten_device_tree()\n");
+
```

Comment:
but, I guess they expected that during init process allocation never failed 'cause the system is fresh a full of free memory